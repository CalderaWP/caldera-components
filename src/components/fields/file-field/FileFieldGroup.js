import React from 'react';
import classNames from 'classnames';
import {RenderGroup} from '../../RenderGroup';
import {Message} from '../messages/Message';
import {FileSelect} from './FileSelect';
import {fieldGroupPropTypes} from '../propTypes';

/**
 * Encapsulates a complete File field group
 */
export class FileFieldGroup extends React.PureComponent {

	/**
	 * Create a FileFieldGroup component
	 *
	 * @param {Object} props
	 */
	constructor(props) {
		super(props);
		this.onChange = this.onChange.bind(this);
		this.onSelect = this.onSelect.bind(this);
	}

	/**
	 * Handle direct change events
	 * @param {String|number} newValue
	 */
	onChange(newValue) {
		this.props.onValueChange(newValue);
	}

	/**
	 * Handle when the option is chosen
	 * @param {String|number} value
	 */
	onSelect(value) {
		this.props.onValueChange(value);
	}

	/**
	 * Render FileFieldGroup component
	 * @return {*}
	 */
	render() {
		return (
			<div
				className={classNames(
					FileFieldGroup.classNames.fieldWrapper,
					RenderGroup.classNames.fieldWrapper,
					this.props.className
				)
				}
			>
			{this.props.message.message &&
				<Message
					message={this.props.message}
				/>
			}
				<label
					htmlFor={this.props.id}
				>{this.props.label}</label>
				
				<FileSelect
					id={this.props.id}
					onValueChange={this.props.onValueChange}
					value={this.props.value}
					isRequired={this.props.isRequired}
					className={this.props.fieldClassName}
					attachToMailer={this.props.attachToMailer}
					saveInLibrary={this.props.saveInLibrary}
				/>
			</div>
		);
	}
}

/**
 * Prop definitions for FileFieldGroup component
 */
FileFieldGroup.propTypes = fieldGroupPropTypes;

/**
 * Default property values for FileFieldGroup component
 *
 * @type {{}}
 */
FileFieldGroup.defaultProps = {
	message: {
		error: false,
		message: ''
	},
	attachToMailer: false,
	saveInLibrary: false,
	type: 'file',
};

/**
 * The names of classes used for HTML elements in FileFieldGroup component
 * @type {{fieldWrapper: string, input: string, option: string}}
 */
FileFieldGroup.classNames = {
	fieldWrapper: 'caldera-file-select-group',
	input: 'caldera-file-input'
};