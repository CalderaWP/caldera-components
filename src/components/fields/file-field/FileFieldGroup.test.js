import renderer from 'react-test-renderer';
import React from 'react';
import {mount, shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {FileFieldGroup} from './FileFieldGroup';

Enzyme.configure({adapter: new Adapter()});

const genericChangeHandler = () => {
};
describe('FileFieldGroup component', () => {


	it('matches snapshot with no message', () => {
		const component = renderer.create(
			<FileFieldGroup
				id={'file-3'}
				label={'Upload file'}
				fieldClassName={'file'}
				onValueChange={genericChangeHandler}
				isOpen={false}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('matches snapshot with message', () => {
		const component = renderer.create(
			<FileFieldGroup
				id={'file-3'}
				label={'Upload file'}
				fieldClassName={'file'}
				onValueChange={genericChangeHandler}
				message={{
					message: 'Hi Roy',
					error: false,
				}}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	describe('Inner input', () => {
		it('Has inner input', () => {
			const component = mount(
				<FileFieldGroup
					id={'file-1'}
					fieldClassName={'file'}
					label={'Upload file'}
					onValueChange={genericChangeHandler}
					isOpen={false}
				/>
			);
			expect(component.find('input')).toHaveLength(1);
		});
	});

	describe('Value', () => {
		it('Passes updated value properly through the onChange handler', () => {
			let updatedValue = null;
			const component = mount(
				<FileFieldGroup
					id={'file-9'}
					fieldClassName={'file'}
					label={'Upload file'}
					onValueChange={(newValue) => {
						updatedValue = newValue;
					}}
					value={'file1.jpg'}
				/>
			);

			component.instance().onChange('file2.jpg');
			expect(updatedValue).toEqual('file2.jpg');
		});
	});

});