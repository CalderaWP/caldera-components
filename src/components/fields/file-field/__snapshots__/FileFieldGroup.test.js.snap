// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileFieldGroup component matches snapshot with message 1`] = `
<div
  className="caldera-file-select-group caldera-config-field"
>
  <div
    className="caldera-components-message caldera-components-not-error"
  >
    Hi Roy
  </div>
  <label
    htmlFor="file-3"
  >
    Upload file
  </label>
  <input
    accept="image/*"
    id="file-3"
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`FileFieldGroup component matches snapshot with no message 1`] = `
<div
  className="caldera-file-select-group caldera-config-field"
>
  
  <label
    htmlFor="file-3"
  >
    Upload file
  </label>
  <input
    accept="image/*"
    id="file-3"
    onChange={[Function]}
    type="file"
  />
</div>
`;
